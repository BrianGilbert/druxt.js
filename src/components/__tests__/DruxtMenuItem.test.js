import { config, createLocalVue, shallowMount,  } from '@vue/test-utils'
import Vuex from 'vuex'
import mockAxios from 'jest-mock-axios'

import { DruxtMenu, DruxtMenuComponent, DruxtMenuItemComponent, DruxtMenuStore } from '../..'

jest.mock('axios')

const baseURL = 'https://demo-api.druxtjs.org'

const propsData = {
  item: {
    entity: {
      attributes: {
        title: 'Parent',
        link: {
          uri: 'internal:/parent'
        }
      }
    },
    children: [{
      entity: {
        attributes: {
          title: 'Child',
          link: {
            uri: 'entity:node/1'
          }
        }
      },
      children: []
    }]
  }
}

const stubs = ['nuxt-link']

// Setup local vue instance.
const localVue = createLocalVue()
localVue.use(Vuex)

const mountComponent = options => {
  config.mocks.$route = {
    path: '/'
  }

  return shallowMount(DruxtMenuItemComponent, {
    localVue,
    propsData,
    parentComponent: DruxtMenuComponent,
    ...options
  })
}

let store

describe('DruxtMenuItem', () => {
  beforeEach(() => {
    // Setup vuex store.
    store = new Vuex.Store()
    store.$druxtMenu = new DruxtMenu(baseURL, {})
    DruxtMenuStore({ store })
    config.mocks.$store = store
  })

  test('default', () => {
    const wrapper = mountComponent()
    // Ensure we get sane default HTML.
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.to).toStrictEqual({
      path: '/parent',
      type: 'internal'
    })
  })

  test('unwrapped', () => {
    const wrapper = mountComponent({ parentComponent: null })
    // Ensure unwrapped instance of `<druxt-menu-item />` doesn't render.
    expect(wrapper.html()).toBe('')
  })
})
