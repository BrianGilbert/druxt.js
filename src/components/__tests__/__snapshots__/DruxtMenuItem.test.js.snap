// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DruxtMenuItem default 1`] = `
"<li class=\\"\\">
  <druxt-menu-item-stub item=\\"[object Object]\\"></druxt-menu-item-stub>
  <ul>
    <druxt-menu-item-stub item=\\"[object Object]\\"></druxt-menu-item-stub>
  </ul>
</li>"
`;
